sdk/identity/identity/src/util/logging.ts

```diff
@@ -1,9 +1,138 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { createClientLogger } from "@azure/logger";
+import { createClientLogger, AzureLogger } from "@azure/logger";
+import { TokenCredentialOptions } from "../client/identityClient";
+import { CredentialUnavailable } from "../client/errors";
 
 /**
  * The AzureLogger used for all clients within the identity package
  */
 export const logger = createClientLogger("identity");
+
+interface EnvironmentAccumulator {
+  missing: string[];
+  assigned: string[];
+}
+
+/**
+ * Separates a list of environment variable names into a plain object with two arrays: an array of missing environment variables and another array with assigned environment variables.
+ * @param supportedEnvVars List of environment variable names
+ */
+export function processEnvVars(supportedEnvVars: string[]): EnvironmentAccumulator {
+  return supportedEnvVars.reduce(
+    (acc: EnvironmentAccumulator, envVariable: string) => {
+      if (process.env[envVariable]) {
+        acc.assigned.push(envVariable);
+      } else {
+        acc.missing.push(envVariable);
+      }
+      return acc;
+    },
+    { missing: [], assigned: [] }
+  );
+}
+
+/**
+ * Based on a given list of environment variable names,
+ * logs the environment variables currently assigned during the usage of a credential that goes by the given name.
+ * Logs nothing if the log level is "warning" or "error".
+ * @param credentialName Name of the credential in use
+ * @param supportedEnvVars List of environment variables supported by that credential
+ */
+export function logEnvVars(credentialName: string, supportedEnvVars: string[]): void {
+  // Logs nothing if the log level is "warning" or "error".
+  const logLevel = process.env.AZURE_LOG_LEVEL;
+  if (logLevel === "warning" || logLevel === "error") {
```

Comment:
I understand that, though I was wondering about performance. This kind of conditionals discard unnecessary processing (though it's probably so little that I'm not sure if we care)

Reply:
I think if generating a log message has negative performance characteristics, we should reconsider that message. Otherwise I think it would be surprising if someone turned on logging to figure out a perf issue and their perf got worse.